implementations/rust/ockam/ockam_command/src/command.rs

```diff
@@ -43,31 +57,198 @@ pub struct OckamCommand {
+        }
+    }
+
+    /// Create the tracing configuration, depending on the command to execute
+    async fn make_tracing_configuration(
+        &self,
+        state: &CliState,
+    ) -> miette::Result<ExportingConfiguration> {
+        if self.subcommand.is_background_node() {
+            ExportingConfiguration::background(state)
+                .await
+                .into_diagnostic()
+        } else {
+            ExportingConfiguration::foreground(state)
+                .await
+                .into_diagnostic()
+        }
+    }
+
     /// Run the command
-    pub fn run(self, arguments: Vec<String>) -> miette::Result<()> {
+    pub async fn run(self, ctx: &mut Context, arguments: &[String]) -> miette::Result<()> {
         // If test_argument_parser is true, command arguments are checked
         // but the command is not executed. This is useful to test arguments
         // without having to execute their logic.
         if self.global_args.test_argument_parser {
             return Ok(());
         }
 
+        load_compile_time_vars();
+
         // Sets a hook using our own Error Report Handler.
         // This allows us to customize how we format the error messages and their content.
         let _hook_result = miette::set_hook(Box::new(|_| Box::new(ErrorReportHandler::new())));
 
         let command_name = self.subcommand.name();
 
-        // Set the in-memory env var if needed
+        let mut in_memory = false;
+
         if let OckamSubcommand::Node(cmd) = &self.subcommand {
             if let crate::node::NodeSubcommand::Create(c) = &cmd.subcommand {
-                if c.in_memory {
-                    std::env::set_var(OCKAM_SQLITE_IN_MEMORY, "true");
-                }
+                in_memory = c.in_memory;
             }
         }
 
-        let options = CommandGlobalOpts::new(&arguments, &self.global_args, &self.subcommand)?;
+        let logging_configuration = self.make_logging_configuration(Term::stdout().is_term())?;
+
+        let (tracing_configuration, tracing_guard, cli_state) = if !is_exporting_set()? {
+            // Allows to have logging enabled before initializing CliState
+            let tracing_configuration = ExportingConfiguration::off().into_diagnostic()?;
+            let tracing_guard =
+                self.setup_logging_tracing(&logging_configuration, &tracing_configuration);
+
+            (tracing_configuration, tracing_guard, None)
+        } else {
+            let cli_state = self.init_cli_state(in_memory).await;
+            let tracing_configuration = self.make_tracing_configuration(&cli_state).await?;
+            let tracing_guard =
+                self.setup_logging_tracing(&logging_configuration, &tracing_configuration);
+            let cli_state = cli_state.set_tracing_enabled(tracing_configuration.is_enabled());
+
+            (tracing_configuration, tracing_guard, Some(cli_state))
+        };
```

Comment:
what are the implications of not initializing the clistate here? Couldn't we initialize it here in both branches and remove the `match cli_state` from below?

Reply:
The idea is to initialize logging as early as possible. The problem is that if tracing is enabled, we unfortunately need CliState to be able to properly bootstrap it. Therefore, I was able to initialize logging before creating the CliState only if tracing is disabled